### PR TITLE
Test: Remove legacy test runner

### PR DESCRIPTION
**Pull request by gziolo**
Part of #3822.

Tasks done:
* [x] Remove all code related to legacy test runner and its references.
* [x] Improve the way `npm test` and `make test` are configured.
* [x] Drop `tests.json` from `client` folder and remove whitelist logic.
* [x] Move files distribution for Circle CI nodes to single test runner to avoid patter matching issues.
* [x] Remove missed `Makefile` files. See https://github.com/Automattic/wp-calypso/issues/3942#issuecomment-212499702.
* [x] Migrate `client/components/feature-example`

### Testing
Run locally:
* `npm test`

Check on Circle CI `Test Summary`:
screen shot 2016-04-21 at 10 04 11

Check on Circle CI if tests are distributed properly between nodes. You can download reports from `Artifacts` tab:
screen shot 2016-04-21 at 10 03 06

### CODE REVIEW

**Comment by gziolo** [comment]
@aduth @blowery: Very similar to #4673. Another cleanup task that we should have ready when all tests are migrated :)

**Comment by gziolo** [comment]
@aduth @blowery I call it the day. I didn't manage to finish this one. There is one more test that needs to be migrated. We have also still issue with Circle CI pattern matching. We have two files included in `client` test runner which have `/abtest/` in path. Any ideas how to tackle it? Should we build our own logic for parallel tests execution?

File: test/runner.js [hunk 1]
```diff
@@ -36,21 +35,9 @@ if ( program.grep ) {
 	mocha.grep( new RegExp( program.grep ) );
 }
 
-if ( process.env.CIRCLECI ) {
```
Comment (gziolo) [inline test/runner.js]: It would be nice to get rid of it, but it looks like some tests are still bombing without it.

**Comment by gziolo** [comment]
/cc @aduth @blowery @mtias @nb 

I removed lots of code we no longer need. I hoped we need less changes to finish unit tests migration, that's why I didn't split this PR into smaller tasks. There aren't many additions so it should be good to review as one batch of changes. Let me know if I should split anything into its own PR.

File: client/components/feature-example/test/index.js [hunk 2]
```diff
@@ -11,14 +11,13 @@ import { shallow } from 'enzyme';
 import FeatureExample from '../index';
 
 describe( 'Feature Example', function() {
-	it( 'should have Feature-example class', function() {
+	it( 'should have Feature-example class', () => {
 		const featureExample = shallow( <FeatureExample /> );
-		assert.equal( 1, featureExample.find( '.feature-example' ).length );
```
Comment (nb) [inline client/components/feature-example/test/index.js]: The expected value is usually first.

Comment (gziolo) [inline client/components/feature-example/test/index.js]: Yeah, [lenghtOf](http://chaijs.com/api/assert/#method_lengthof) is tricky.

Comment (nb) [inline client/components/feature-example/test/index.js]: Yeah, I get it now – the grammar would be wrong if the length was the first argument.

**Comment by Tug** [comment]
Nice work here. All tests passed and ran in 25s on my computer!
Code LGTM 👍

**Comment by Tug** [comment]
Would be nice to be able to run a subset of the tests in a next PR. I tried the `-g LoginTest` option and it did work if I call the runner directly but not from `npm test`. Would be call to watch a directory also.

**Comment by gziolo** [comment]
@Tug Thanks for review and all feedback. Great ideas, feel free to add them to the list [here](https://github.com/Automattic/wp-calypso/issues/4915) 👍

**Comment by nb** [comment]
🎉

**Pull request merged**